cmd/kubeadm/app/preflight/checks.go

```diff
@@ -325,6 +337,112 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return nil, nil
 }
 
+type etcdVersionResponse struct {
+	Etcdserver  string
+	Etcdcluster string
+}
+
+// ExternalEtcdVersionCheck checks if version of external etcd meets the demand of kubeadm
+type ExternalEtcdVersionCheck struct {
+	Etcd kubeadmapi.Etcd
+}
+
+func (evc ExternalEtcdVersionCheck) Check() (warnings, errors []error) {
+	var config *tls.Config
+	var err error
+	if config, err = evc.configRootCAs(config); err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+	if config, err = evc.configCertAndKey(config); err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	client := evc.getHTTPClient(config)
+	for _, endpoint := range evc.Etcd.Endpoints {
+		resp := etcdVersionResponse{}
+		if strings.HasSuffix(endpoint, "/") {
```

Comment:
golang net/url packages seems not have appropriate method for http url concatenating and cleaning, but kubernetes has already vendored `github.com/PuerkitoBio/purell`, so i use it to normalize urls